source/sessions/snapshot-sessions.rst

```diff
@@ -0,0 +1,353 @@
+        * ``session = client.startSession(isSnapshot = true)``
+        * ``document = collection.anyReadOperation(session, ...)``
+        * capture the command sent to the server (using APM or other mechanism)
+        * assert that the command does not have an ``atClusterTime``
+
+2.  | Subsequent snapshot reads on a ``ClientSession`` should read from the snapshot of the first read in that session.
+    | Test SHOULD introduce two variations one for primaries and additional for secondary reads.
+        * ``session1 = client.startSession(isSnapshot = true)``
+        * ``session2 = client.startSession(isSnapshot = true)``
+        * ``readBeforeUpdateSession1 = collection.anyReadOrOperation(session1, ...)``
+        * ``collection.anyUpdateOpertation(...)``
+        * ``readBeforeUpdateSession2 = collection.anyReadOrOperation(session2, ...)``
+        * ``collection.anyUpdateOpertation(...)``
+        * ``readAfterUpdateSession1 = collection.anyReadOrOperation(session1, ...)``
+        * ``readAfterUpdateSession2 = collection.anyReadOrOperation(session2, ...)``
+        * Assert that `readBeforeUpdateSession1` is equivalent to `readAfterUpdateSession1` and 
+        `readBeforeUpdateSession2` to `readAfterUpdateSession2`
+
+3.  | Snapshot reads that fail with ``SnapshotError`` are not retried.
+    | This test is OPTIONAL as it requires setting ``session.snapshotClusterTime`` which is not part of public API.
+        * ``session = client.startSession(isSnapshot = true)``
+        * ``session.snapshotClusterTime = 0``
+        * ``collection.anyReadOrOperation(session1, ...)``
+        * Assert that only single read command was issued
+
+4.  | A read operation in a ``ClientSession`` that is not snapshot read
+    | should not include the ``atClusterTime`` parameter in the command sent to the
+    | server
+        * ``session = client.startSession(isSnapshot = false)``
+        * ``collection.anyReadOperation(session, {})``
+        * ``operationTime = session.operationTime``
+        * capture the command sent to the server (using APM or other mechanism)
+        * assert that the command does not have an ``atClusterTime`` field
+
+Motivation 
+==========
+
+To support snapshot reads. Only supported with server version 4.9 or newer.
+
+Design Rationale
+================
+
+The goal is to modify the driver API as little as possible so that existing
+programs that don't need snapshot reads don't have to be changed.
+This goal is met by defining a ``SessionOptions`` field that applications use to
+start a ``ClientSession`` that can be used for snapshot reads.
+
+Backwards Compatibility
+=======================
+
+The API changes to support sessions extend the existing API but do not
+introduce any backward breaking changes. Existing programs that don't use
+snapshot reads continue to compile and run correctly.
+
+Reference Implementation
+========================
+
+A reference implementation must be completed before any spec is given status
```

Comment:
Update to indicate that C# is the reference implementation.

Reply:
Done.

Reply:
I'm not seeing any changes to the PR.  Did you not push yet?

Reply:
Oh, I see it now.  The intention though, I think, is to remove all of the flowery prose and just have the one sentence saying which driver is the reference implementation.

Reply:
Done.